pkg/deploy/template_chart_manager.go

```diff
@@ -0,0 +1,196 @@
+	SkrWebhookMemoryLimits     string
+	SkrWebhookCPULimits        string
+	WatcherLocalTestingEnabled bool
+	GatewayHTTPPortMapping     int
+}
+
+func NewSKRWebhookTemplateChartManager(kcpRestConfig *rest.Config, config *SkrChartManagerConfig,
+) (*SKRWebhookTemplateChartManager, error) {
+	resolvedKcpAddr, err := resolveKcpAddr(kcpRestConfig, config)
+	if err != nil {
+		return nil, err
+	}
+	return &SKRWebhookTemplateChartManager{
+		config:  config,
+		cache:   &sync.Map{},
+		kcpAddr: resolvedKcpAddr,
+	}, nil
+}
+
+func (m *SKRWebhookTemplateChartManager) renderChartToRawManifest(ctx context.Context, kymaObjKey client.ObjectKey,
+	chartArgValues map[string]interface{},
+) (string, error) {
+	chartFS := os.DirFS(m.config.WebhookChartPath)
+	chart, err := helm.LoadChart(ctx, chartFS)
+	if err != nil {
+		return "", nil
+	}
+	return helm.Template(ctx, helm.TemplateConfig{
+		Chart:       chart,
+		ReleaseName: skrChartReleaseName(kymaObjKey),
+		Namespace:   metav1.NamespaceDefault,
+		Values:      chartArgValues,
+	})
+}
+
+func (m *SKRWebhookTemplateChartManager) cachedConfig(kymaObjKey client.ObjectKey,
+	chartArgValues map[string]interface{},
+) bool {
+	cachedArgValues, ok := m.cache.Load(kymaObjKey)
+	if !ok {
+		m.cache.Store(kymaObjKey, chartArgValues)
+		return false
+	}
+	if !reflect.DeepEqual(chartArgValues, cachedArgValues) {
+		m.cache.Store(kymaObjKey, chartArgValues)
+		return false
+	}
+	return true
+}
+
+func (m *SKRWebhookTemplateChartManager) Install(ctx context.Context, kyma *v1alpha1.Kyma) (bool, error) {
+	logger := logf.FromContext(ctx)
+	kymaObjKey := client.ObjectKeyFromObject(kyma)
+	syncContext := remote.SyncContextFromContext(ctx)
+	chartArgValues, err := generateHelmChartArgs(ctx, syncContext.ControlPlaneClient, m.config, m.kcpAddr)
+	if err != nil {
+		return true, err
+	}
+	if m.cachedConfig(kymaObjKey, chartArgValues) {
+		logger.V(int(zap.DebugLevel)).Info("webhook chart config is already installed",
+			"release-name", skrChartReleaseName(kymaObjKey))
+		return false, nil
+	}
+	manifest, err := m.renderChartToRawManifest(ctx, kymaObjKey, chartArgValues)
+	if err != nil {
+		return true, err
+	}
+	resources, err := getRawManifestUnstructuredResources(manifest)
+	if err != nil {
+		return true, err
+	}
+	for _, resource := range resources {
+		resourceObjKey := client.ObjectKeyFromObject(resource)
+		oldResource := &unstructured.Unstructured{}
+		oldResource.SetGroupVersionKind(resource.GroupVersionKind())
+		err := syncContext.RuntimeClient.Get(ctx, resourceObjKey, oldResource)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return true, fmt.Errorf("failed to get webhook %s: %w", resource.GetKind(), err)
+		}
+		if apierrors.IsNotFound(err) {
+			if err := syncContext.RuntimeClient.Create(ctx, resource, defaultFieldOwner); err != nil {
+				return true, fmt.Errorf("failed to create webhook %s: %w", resource.GetKind(), err)
+			}
+		}
+		// completely replace old resource with new resource
```

Comment:
considering that the resource was freshly created, why is it being updated right after. Somehow this doesn't make sense.